src/utils.lua

```diff
@@ -71,15 +88,17 @@ function M.build_luarocks(luarocks_dir, prefix)
   lfs.chdir(current_dir)
 end
 
-function M.write_activate_script(template, lua_version, prefix)
-  local activate_file, err = io.open(prefix.."/bin/activate", "w+")
+function M.write_activate_script(template, params)
+  local activate_file, err = io.open(params.prefix.."/bin/activate", "w+")
 
   if not activate_file then
     print("Failed to open activate file: "..err)
     os.exit(3)
   end
 
-  activate_file:write(string.format(template, lua_version, prefix))
+  activate_file:write((template:gsub("%%{([%w_]+)}", function(var)
```

Comment:
instead of `%s`, this replaces `%{var_name}` with the content of `params.var_name` (or an empty string if not defined)